pkg/apis/core/validation/validation.go

```diff
@@ -4515,8 +4598,11 @@ func ValidatePodTemplate(pod *core.PodTemplate, opts PodValidationOptions) field
 // that cannot be changed.
 func ValidatePodTemplateUpdate(newPod, oldPod *core.PodTemplate, opts PodValidationOptions) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPod.ObjectMeta, &oldPod.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidatePodTemplateSpec(&newPod.Template, field.NewPath("template"), opts)...)
-	return allErrs
+	newPodClone := newPod.DeepCopy()
+	CleanPodTemplateSpec(&newPodClone.Template)
```

Comment:
Here you only have to clone the templatespec, not the whole pod.

Reply:
done